bindings/python/visualize/meshcat_visualizer.py

```diff
@@ -0,0 +1,264 @@
+                                       idyn.Transform.Identity(), joint_pos,
+                                       self.link_pos[model_name])
+
+        link_solid_shapes = model_geometry.getLinkSolidShapes()
+
+        for link_index in range(0, self.model[model_name].getNrOfLinks()):
+
+            world_H_frame = self.link_pos[model_name](link_index)
+            link_name = self.model[model_name].getLinkName(link_index)
+
+            is_mesh = False
+            for geom in range(0, len(link_solid_shapes[link_index])):
+                solid_shape = model_geometry.getLinkSolidShapes()[link_index][geom]
+                if self.__is_mesh(solid_shape):
+                    obj = self.__load_mesh(solid_shape)
+                    is_mesh = True
+                else:
+                    msg = "The geometry object named " \
+                          + solid_shape.asExternalMesh().getName() \
+                          + " is not supported by iDynTree/MeshCat for visualization."
+                    warnings.warn(msg, category=UserWarning, stacklevel=2)
+                    continue
+
+                if obj is None:
+                    msg = "The geometry object named " + solid_shape.asExternalMesh().getName() + " is not valid."
+                    warnings.warn(msg, category=UserWarning, stacklevel=2)
+                    continue
+
+                viewer_name = model_name + "/" + link_name + "/" + solid_shape.asExternalMesh().getName()
+
+                if isinstance(obj, meshcat.geometry.Object):
+                    self.viewer[viewer_name].set_object(obj)
+                elif isinstance(obj, meshcat.geometry.Geometry):
+                    material = meshcat.geometry.MeshPhongMaterial()
+                    # Set material color from URDF, converting for triplet of doubles to a single int.
+                    if color is None:
+                        mesh_color = solid_shape.getMaterial().color()
+                    else:
+                        mesh_color = color
+
+                    material.color = int(mesh_color[0] * 255) * 256 ** 2 + \
+                                     int(mesh_color[1] * 255) * 256 + \
+                                     int(mesh_color[2] * 255)
+
+                    # Add transparency, if needed.
+                    if float(mesh_color[3]) != 1.0:
+                        material.transparent = True
+                        material.opacity = float(mesh_color[3])
+
+                    self.viewer[viewer_name].set_object(obj, material)
+
+                    if is_mesh:
+                        self.__apply_transform(world_H_frame, solid_shape, viewer_name)
+
+    def display(self, base_position, base_rotation, joint_value, model_name='iDynTree'):
+        """Display the robot at given configuration."""
+
+        if not self.__model_exists(model_name):
+            msg = "The model named: " +  model_name + " already exists."
```

Comment:
Is the if checking if the model exists or not?

Reply:
Actually here I was checking if the model is already registered (or loaded) in the visualizer. I think I should change the name of the function. Do you have any suggestions?

Reply:
I think the name of the function is ok. It is the message that is probably wrong. I suppose you want to notify the user if the model does not exists, am I correct?

Reply:
ops I noticed just now. sorry for the typo

Reply:
https://github.com/robotology/idyntree/pull/901/commits/bc77b0174df8a9639faae68bf20ed75db832943d